### DISCUSSION TITLE
Route Model Binding and Soft Delete

### DISCUSSION THREAD

**Discussion by mrfelipemartins** [livewire/livewire#7976] [1. Help] [answered]
I have a problem using Route Model Binding for soft deleted models. Livewire always throws 404. I tried customizing the resolution logic in RouteServiceProvider since I am using full page components but it seems ignored by Livewire somehow. So I came up with a workaround by adding the following to my models:

```php
public function resolveRouteBinding($value, $field = null)
    {
        if ($field === null) {
            $field = $this->getKeyName();
        }

        if (method_exists($this, 'getOptimus')) {
            $value = $this->getOptimus()->decode($value);
        }

        $query = $this->where($field, $value);

        if (method_exists($this, 'bootSoftDeletes')) {
            $query->withoutGlobalScope(SoftDeletingScope::class);
        }

        return $query->firstOrFail();
    }
```

Is this an actual bug or is there some other way to customize the binding logic used by Livewire?

**Comment by joshhanley** [comment, marked as answer]
@mrfelipemartins This has been fixed recently in PR #7913 and tagged in todays release 🙂

**Reply by mrfelipemartins** [reply]
Awesome! I missed that. Thank you.

**Reply by joshhanley** [reply]
@mrfelipemartins No worries! 🙂

**Reply by mrfelipemartins** [reply]
@joshhanley but what about this example:

```php
public function restore(User $user): void
{
    $user->restore();
}

<button wire:click="restore({{$user->id}})">Restore</button>
```

In this case I would still see a 404, right? How this should be handled within Livewire? Currently I am using the workaround I provided above.

**Reply by joshhanley** [reply]
@mrfelipemartins dependency injection is different, as most people wouldn't expect deleted models to be resolved by the container.

You could keep your work around as long as you keep in mind that a resolved model can be both deleted and not deleted. But personally I would just change my Livewire method to something like this:

```php
public function restore(int $userId): void
{
    $user = User::onlyTrashed()->find($userId);
    $user->restore();
}
```

**Reply by mrfelipemartins** [reply]
That does make sense, thanks for the clarification!

**Reply by joshhanley** [reply]
No worries! 🙂
